pkg/cmd/workspace/ssh.go

```diff
@@ -98,12 +101,24 @@ var SshCmd = &cobra.Command{
 			sshArgs = append(sshArgs, args[2:]...)
 		}
 
+		sshOptions := make(map[string]string)
+		for _, option := range sshOptionsMap {
```

Comment:
From what I can see, you first convert an array to a map and the convert it back to an array in `OpenTerminalSsh`

Simply pass the array to `OpenTerminalSsh` and then perform the validation there.

Additionally, if a bad argument is passed to `ssh`, it fails. You only log a warning to the user.
We should change that to be consistent with `ssh`.
![Screenshot 2024-10-21 at 10 09 59](https://github.com/user-attachments/assets/37613d7b-e652-41ed-8477-6a991e72fa63)

Reply:
did you mean that we should also fail just like ssh and only log a warning to the user.

Reply:
> did you mean that we should also fail just like ssh

If I run `daytona ssh -o invalid` it should fail because it's not key=value.

Reply:
done